cli/doctest_runner.rs

```diff
@@ -0,0 +1,439 @@
+  lazy_static! {
+    static ref JS_DOC_PATTERN: Regex =
+      Regex::new(r"/\*\*\s*\n([^\*]|\*[^/])*\*/").unwrap();
+    // IMPORT_PATTERN doesn't match dynamic imports by design
+    static ref IMPORT_PATTERN: Regex =
+      Regex::new(r"import[^(].*\n").unwrap();
+    static ref EXAMPLE_PATTERN: Regex = Regex::new(r"@example\s*(?:<\w+>.*</\w+>)*\n(?:\s*\*\s*\n*)*```").unwrap();
+    static ref TEST_TAG_PATTERN: Regex = Regex::new(r"@example\s*(?:<\w+>.*</\w+>)*\n(?:\s*\*\s*\n*)*```(\w+)").unwrap();
+    static ref AWAIT_PATTERN: Regex = Regex::new(r"\Wawait\s").unwrap();
+  }
+
+  let mut import_set = HashSet::new();
+
+  let test_bodies = JS_DOC_PATTERN
+    .captures_iter(&input)
+    .filter_map(|caps| caps.get(0).map(|c| (c.start(), c.as_str())))
+    .flat_map(|(offset, section)| {
+      EXAMPLE_PATTERN.find_iter(section).filter_map(move |cap| {
+        section[cap.end()..].find("```").map(|i| {
+          (
+            offset + cap.end(),
+            section[cap.start()..i + cap.end()].to_string(),
+          )
+        })
+      })
+    })
+    .filter_map(|(offset, example_section)| {
+      let test_tag = TEST_TAG_PATTERN
+        .captures(&example_section)
+        .and_then(|m| m.get(1).map(|c| c.as_str()));
+
+      if test_tag == Some("text") {
+        return None;
+      }
+
+      IMPORT_PATTERN
+        .captures_iter(&example_section)
+        .filter_map(|caps| caps.get(0).map(|m| m.as_str()))
+        .for_each(|import| {
+          import_set.insert(import.to_string());
+        });
+
+      let caption = get_caption_from_example(&example_section);
+      let line_number = &input[0..offset].lines().count();
+      let code_block = get_code_from_example(&example_section);
+      let is_async = AWAIT_PATTERN.find(&example_section).is_some();
+
+      let cwd = std::env::current_dir()
+        .expect("expected: process has a current working directory");
+      let path = p
+        .to_str()
+        .map(|x| x.replace(cwd.to_str().unwrap_or(""), ""));
+      Some(DocTestBody {
+        caption,
+        line_number: *line_number,
+        path: path.unwrap_or_else(|| "".to_string()),
+        value: code_block,
+        ignore: test_tag == Some("ignore"),
+        is_async,
+      })
+    })
+    .collect::<Vec<_>>();
+
+  match test_bodies.len() {
+    0 => None,
+    _ => Some(DocTest {
+      imports: import_set,
+      bodies: test_bodies,
+    }),
+  }
+}
+
+pub fn render_doctest_file(
+  doctests: Vec<DocTest>,
+  fail_fast: bool,
+  quiet: bool,
+  filter: Option<String>,
+) -> String {
+  let mut test_file = "".to_string();
+
+  // TODO(iykekings) - discuss with team if this is fine
+  let default_import = "import { 
+    assert,
+    assertArrayContains,
+    assertEquals,
+    assertMatch,
+    assertNotEquals,
+    assertStrContains,
+    assertStrictEq,
+    assertThrows,
+    assertThrowsAsync,
+    equal,
+    unimplemented,
+    unreachable,
+   } from \"https://deno.land/std/testing/asserts.ts\";\n";
```

Comment:
Not great, but should be fine for first pass. Please interpolate URL with current version of Deno (`crate::version`)

Reply:
`crate::version` doesn't provide std version? Where can I get the std version?